commands/shell.go

```diff
@@ -0,0 +1,89 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/c-bata/go-prompt"
+	"github.com/ionos-cloud/ionosctl/v6/internal/client"
+	"github.com/ionos-cloud/ionosctl/v6/internal/core"
+	"github.com/ionoscloudsdk/comptplus"
+	"github.com/spf13/viper"
+)
+
+var advancedPrompt = &comptplus.CobraPrompt{
+	RootCmd:                  rootCmd.Command,
+	ShowHelpCommandAndFlags:  true,
+	DisableCompletionCommand: true,
+	AddDefaultExitCommand:    true,
+	ShowHiddenCommands:       true,
+	ShowHiddenFlags:          true,
+	GoPromptOptions: []prompt.Option{
+		prompt.OptionTitle("ionosctl"),
+		prompt.OptionPrefix("> "),
+		prompt.OptionShowCompletionAtStart(),
+
+		prompt.OptionAddKeyBind(prompt.KeyBind{Key: prompt.ShiftLeft, Fn: prompt.GoLeftWord}),
+		prompt.OptionAddKeyBind(prompt.KeyBind{Key: prompt.ShiftRight, Fn: prompt.GoRightWord}),
+		prompt.OptionAddKeyBind(prompt.KeyBind{Key: prompt.ShiftDown, Fn: prompt.GoLineBeginning}),
+		prompt.OptionAddKeyBind(prompt.KeyBind{Key: prompt.ShiftUp, Fn: prompt.GoLineEnd}),
+		prompt.OptionAddKeyBind(prompt.KeyBind{Key: prompt.ShiftDelete, Fn: prompt.DeleteWord}),
```

Comment:
Same thing as above, but I think this might require a bit more work to "fix". You'd expect `Delete` to erase characters to the right of the cursor, not the left. From what I saw, there is no `ShiftBackspace` or `ControlBackspace` so that the shortcut could fit the functionality. Perhaps we could implement a replacement for `prompt.DeleteWord` so that the word to the right of the cursor will be deleted?

Reply:
We can just leave the default go-prompt controls I guess, and return to this custom controls topic at a later date when we can allow the user to customize his controls.